File: Javascript/binarySearch.js
let iterativeFunction = function (arr, x) {
  
    let start = 0 
    let end = arr.length-1
         
    while (start<=end){
 
        let mid = Math.floor((start + end)/2);
  
        if (arr[mid]===x) return true;
 
        else if (arr[mid] < x)
             start = mid + 1;
        else
             end = mid - 1;
    }
  
    return false;
}
  
let arr = [1, 3, 5, 7, 8, 9];
let x = 5;
  
if (iterativeFunction(arr, x))
    console.log('Element found!')
else {
    console.log('Element not found!')
}

File: C/binary_search.c
#include<stdio.h>
#include<conio.h>
void main()
{
int a[10],i,n,low=0,high,mid,item;
clrscr();
printf("Enter the number of elements in the array);
scanf("%d",&n);
printf("Enter the elements in the array");
for(i=0;i<n;i++)
  scanf("%d",&a[i]);
printf("Enter the element to be searched");
scanf("%d",&item);
low=0;
high=n-1;
while(low<=high)
{
mid=(low+high)/2;
if(item==a[mid])
{
printf("The element is found at position %d",mid+1);
break;
}
else if(item<a[mid])
  high=--mid;
else
  low=++mid;
}
if(low>high)
  printf("The element is not found");
getch();
}
